src/main/java/org/folio/circulation/resources/BatchChangeDueDateResource.java

```diff
@@ -0,0 +1,85 @@
+package org.folio.circulation.resources;
+
+import org.folio.circulation.domain.ChangeDueDateRepository;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.representations.changeduedate.BatchChangeDueDateRequest;
+import org.folio.circulation.domain.representations.changeduedate.BatchChangeDueDateRequestValidator;
+import org.folio.circulation.domain.representations.changeduedate.BatchChangeDueDateResponseMapper;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.RouteRegistration;
+import org.folio.circulation.support.http.server.WebContext;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CopyOnWriteArrayList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BatchChangeDueDateResource extends Resource {
+
+  private final String rootPath;
+  private static final Logger log = LoggerFactory
+    .getLogger(MethodHandles.lookup().lookupClass());
+
+  public BatchChangeDueDateResource(HttpClient client, String rootPath) {
+    super(client);
+    this.rootPath = rootPath;
+  }
+
+  @Override
+  public void register(Router router) {
+
+    RouteRegistration routeRegistration = new RouteRegistration(
+      rootPath, router);
+
+    routeRegistration.create(this::batchChangeDueDate);
+  }
+
+  private void batchChangeDueDate(RoutingContext routingContext) {
+
+    final WebContext context = new WebContext(routingContext);
+    final Clients clients = Clients.create(context, client);
+    ChangeDueDateRepository changeDueDateRepository = new ChangeDueDateRepository(
+      clients);
+    List<String> failedIds = new CopyOnWriteArrayList<>();
+
+    Result<BatchChangeDueDateRequestValidator> requestValidator = new BatchChangeDueDateRequestValidator(
+      routingContext).validate();
+    if (requestValidator.failed()) {
+      requestValidator.cause().writeTo(routingContext.response());
+      return;
+    }
+
+    BatchChangeDueDateRequest request = requestValidator.value()
+      .createRequest();
+    CompletableFuture.allOf(
```

Comment:
This appears to submit all of the change due date operations singularly in parallel. It seems likely that this would cause similar scalability issues to the parallel submission by the client. How is this different?